src/AutoPilotPlugins/APM/APMSensorsComponent.cc

```diff
@@ -117,7 +117,8 @@ bool APMSensorsComponent::compassSetupNeeded(void) const
     for (size_t i=0; i<cCompass; i++) {
         if (_autopilot->getParameterFact(FactSystem::defaultComponentId, devicesIds[i])->rawValue().toInt() != 0) {
             for (size_t j=0; j<cOffset; j++) {
-                if (_autopilot->getParameterFact(FactSystem::defaultComponentId, rgOffsets[i][j])->rawValue().toFloat() == 0.0f) {
```

Comment:
0.0 is set directly as a default for the parameter by the firmware. We are checking to see if the value has been changed by the user to anything other than the default. Hence comparison to exact 0.0 is what we want here.